src/test/kotlin/com/idan/pokemon_hub/controller/PokemonControllerTest.kt

```diff
@@ -0,0 +1,121 @@
+package com.idan.pokemon_hub.controller
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.idan.pokemon_hub.PokemonController
+import com.idan.pokemon_hub.model.Pokemon
+import com.idan.pokemon_hub.service.PokemonService
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doNothing
+import org.mockito.kotlin.whenever
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.http.MediaType
+import org.springframework.test.context.bean.override.mockito.MockitoBean
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+import java.net.URI
+
+@WebMvcTest(PokemonController::class)
+class PokemonControllerTest {
+
+    @Autowired
+    private lateinit var mockMvc: MockMvc
+
+    @MockitoBean
+    private lateinit var pokemonService: PokemonService
+
+    private val objectMapper = jacksonObjectMapper()
+    private val baseUrl = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/"
+    private val testPokemon = Pokemon(1, "Bulbasaur", "grass/poison", URI("$baseUrl/1.png").toURL())
+
+    @BeforeEach
+    fun setUp() {
+        whenever(pokemonService.getAll()).thenReturn(listOf(testPokemon))
+
+        whenever(pokemonService.geByPokedex(any())).thenAnswer { invocation ->
+            val id = invocation.getArgument<Long>(0)
+            if (id == 1L) testPokemon else null
+        }
+
+        whenever(pokemonService.updateByPokedex(any(), any())).thenAnswer { invocation ->
+            val id = invocation.getArgument<Long>(0)
+            if (id == 1L) testPokemon.copy(name = "Updated Name") else null
+        }
+
+        doNothing().whenever(pokemonService).deleteByPokedex(any())
+    }
+
+
+    @Test
+    @DisplayName("should return all pokemons when GET /pokemons is called")
+    fun shouldReturnAllPokemonsWhenGetAllIsCalled() {
+        // when + then
+        mockMvc.perform(get("/pokemons"))
+            .andExpect(status().isOk)
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(jsonPath("$[0].pokedex").value(1))
+            .andExpect(jsonPath("$[0].name").value("Bulbasaur"))
+            .andExpect(jsonPath("$[0].type").value("grass/poison"))
+    }
+
+    @Test
+    @DisplayName("should return pokemon when GET /pokemons/{pokedex} is called with valid id")
+    fun shouldReturnPokemonWhenValidPokedexIsProvided() {
```

Comment:
We're in kotlin, it's better to use backticks for writing test names so they become more readable. 

```suggestion
    fun `should return pokemon when valid pokedex is provided`() {
```

Reply:
refactored all tests to backticks-based titles.